tests/test_api.py

```diff
@@ -556,6 +556,17 @@ def run():
         t.start()
         t.join()
 
+def test_pool_close():
+    pool = ConnectionPool(size=3, **connection_kwargs)
+
+    with pool.connection():
+        with assert_raises(TypeError):
+            pool.close_connections(timeout='foo')
+
+        with assert_raises(ValueError):
+            pool.close_connections(timeout=-1)
+
+        pool.close_connections()
```

Comment:
me either :) ... I thought about mocking the connections, but that's a new pattern to the tests. as a result I wasn't sure it was a good path to go down.